meshtastic/mesh_interface.py

```
@@ -1303,12 +1785,11 @@ def sendTelemetry(
         telemetry_type = telemetryType
         if telemetry_type not in VALID_TELEMETRY_TYPE_SET:
             # Backwards compatibility: unknown types fall back to device_metrics with deprecation warning
-            warnings.warn(
+            logger.warning(
                 f"Unsupported telemetryType '{telemetry_type}' is deprecated. "
                 f"Supported values: {sorted(VALID_TELEMETRY_TYPES)}. "
                 f"Falling back to '{DEFAULT_TELEMETRY_TYPE}'. "
                 "This will raise an error in a future version.",
```

Ruff (0.15.4)

[warning] Logging statement uses f-string

(G004)